MFVNeutralino/plugins/AnalysisCuts.cc

```diff
@@ -479,379 +589,572 @@ bool MFVAnalysisCuts::filter(edm::Event& event, const edm::EventSetup& setup) {
+            if(sel_btags_hard < 2) return false;
 
-                        for(int j2 = j1+1; j2 < njets; ++j2){
-                            if(!jet_hlt_match(mevent, j2) || mevent->jet_pt[j2] < 35) continue;
+            for(int j0 = 0; j0 < njets; ++j0){
+                if(!jet_hlt_match(mevent, j0) || mevent->jet_pt[j0] < 110) continue;
 
-                            for(int j3 = j2+1; j3 < njets; ++j3){
-                                if(!jet_hlt_match(mevent, j3) || mevent->jet_pt[j3] < 35) continue;
+                for(int j1 = j0+1; j1 < njets; ++j1){
+                    if(!jet_hlt_match(mevent, j1) || mevent->jet_pt[j1] < 110) continue;
 
-                                passed_kinematics = true;
-                            }
-                        }
+                    if(fabs(mevent->jet_eta[j0] - mevent->jet_eta[j1]) < 1.6){
+                        passed_kinematics = true;
                     }
                 }
-                return passed_kinematics;
             }
+            return passed_kinematics;
+        }
+    case mfv::b_HLT_PFMET120_PFMHT120_IDTight :
+        {
+         //if(mevent->met() < 150 || njets < 2) return false; // cut on MET to avoid turn-on region, maybe cut value need to be determined
+         //if (njets < 2) return false;
+            return true;
+        }
+
+    default :
+        {
+            throw std::invalid_argument(std::string(mfv::hlt_paths[trig]) + " not implemented in satisfiesTrigger");
+        }
+  }
+  return false;
+}
+
 
-        case mfv::b_HLT_DoubleJetsC100_DoubleBTagCSV_p014_DoublePFJetsC100MaxDeta1p6 :
-            {
-                if(year != 20161 and year != 20162) return false;
-                if(sel_btags_hard < 2) return false;
+//splitting up the satisfieslep trigger into muon/ele trigger 
+bool MFVAnalysisCuts::satisfiesLepTrigger(edm::Handle<MFVEvent> mevent, size_t trig, const edm::EventSetup& setup) { 
+  if(!mevent->pass_hlt(trig)) return false;
+
+  int nmuons     = mevent->nmuons();
+  int nelectrons = mevent->nelectrons();
+  int njets      = mevent->njets(20);
+    
+  bool passed_kinematics = false;
+
+  switch(trig){
+  case mfv::b_HLT_Ele35_WPTight_Gsf : //For 2017
+  //case mfv::b_HLT_Ele32_WPTight_Gsf : //For 2018
+    {
+      for(int ie =0; ie < nelectrons; ++ie){
+        //if (mevent->electron_pt[ie] < 35) continue; //For 2018
+	if (mevent->electron_pt[ie] < 38) continue; //For 2017
+	      if (mevent->electron_ID[ie][3] == 1) {
+	        if (abs(mevent->electron_eta[ie]) < 2.4) { 
+	          if (mevent->electron_iso[ie] < 0.10) {
+	            passed_kinematics = true;
+	          } 
+	        }
+	      }
+      }
+      return passed_kinematics;
+    }
+  case mfv::b_HLT_IsoMu27 : //For 2017
+  //case mfv::b_HLT_IsoMu24 : //For 2018
+    {
+      for(int im =0; im < nmuons; ++im) {
+	//if (mevent->muon_pt[im] < 27) continue; //For 2018
+          if (mevent->muon_pt[im] < 30) continue; //For 2017
```

Comment:
Would be good if these did not have to be commented / uncommented (too easy to make a mistake). @abbywarden @pkotamnives. Same question from before: does @duquale need to rerun his jobs submitted for a different year, to account for this in the Kshort study (that relies on lepton triggers)?

Reply:
Same comment as TriggerEnum and TriggerFilter_cfi -- this has been cleaned up so as not to mess up switching between years, just need to include it in this pull request